packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -701,9 +702,9 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
 		}
 	}
 
-	private disconnectHandler(reason: string) {
+	private disconnectHandler(reason: string, errorMessage?: string) {
 		this.messageBuffer.length = 0;
-		this.emit("disconnect", reason);
+		this.emit("disconnect", { reason, errorMessage });
```

Comment:
This part of change allows side-observers to see the details. But without any changes in Container.ts, all this change does is logging "Closing DeltaManager" and dropping the rest of the details.

There are two options here - either add a change in Container.ts (see "disconnect" handler) and keep propagating it through ConnectionStateHandler to logConnectionStateChangeTelemetry().

OR drop most of the changes in this PR and just leave this change only:
```ts
const disconnectReason = "Closing DeltaManager";
```
I think second one is preferred. Mostly because disconnect events are more about connections themselves, they are not about external errors (in app code). So when we are closing container, all we want to communicate about connection is that we are closing it because container is closed.
If someone wants to know why container is closed, they do have access to Container's "closed" event and "ContainerClose" telemetry event that do communicate this extra reason.